.github/workflows/ci.yml

```diff
@@ -9,61 +9,58 @@ on:
-
-  test-features-default-with-serde:
-    name: "Test Suite [Features: Default + serde]"
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - uses: dtolnay/rust-toolchain@stable
-      - run: cargo build --features serde
-      - run: cargo test --features serde
-
-  test-features-alloc:
-    name: "Test Suite [Features: alloc]"
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v3
-      - uses: dtolnay/rust-toolchain@stable
-      - run: cargo build --no-default-features --features alloc,grid
-      - run: cargo test --no-default-features --features alloc,grid
-
-  test-features-alloc-no-grid:
-    name: "Test Suite [Features: alloc (no grid)]"
+      - uses: Leafwing-Studios/cargo-cache@v1
+      - run: |
+          cargo build
+          cargo test
+  
+  # This tests all enabled default features with each of the optional features on top.
+  test-features-with-default:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        feature: [serde, random, alloc]
     steps:
       - uses: actions/checkout@v3
       - uses: dtolnay/rust-toolchain@stable
-      - run: cargo build --no-default-features --features alloc
-      - run: cargo test  --no-default-features --features alloc
-
-  test-features-default-no-grid:
-    name: "Test Suite [Features: std (no grid)]"
+      - uses: Leafwing-Studios/cargo-cache@v1
+      - run: |
+          cargo build --features ${{ matrix.feature }}
+          cargo test --features ${{ matrix.feature }}
+  
+  # This test all layout features with each of the optional features in isolation.
+  test-features-without-default:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        layout: [flexbox, grid, none]
+        feature: [std, alloc, serde, none]
     steps:
       - uses: actions/checkout@v3
       - uses: dtolnay/rust-toolchain@stable
       - uses: Leafwing-Studios/cargo-cache@v1
-      - run: cargo build --no-default-features --features std
-      - run: cargo test --no-default-features --features std
+      - if: ${{ matrix.layout == 'none' && matrix.feature == 'none' }}
+        run: |
+          cargo build --no-default-features
+          cargo test --no-default-features
+      - if: ${{ matrix.layout != 'none' && matrix.feature == 'none' }}
+        run: |
+          cargo build --no-default-features --features ${{ matrix.layout }}
+          cargo test --no-default-features --features ${{ matrix.layout }}
+      - if: ${{ matrix.layout == 'none' && matrix.feature != 'none' }}
+        run: |
+          cargo build --no-default-features --features ${{ matrix.feature }}
+          cargo test --no-default-features --features ${{ matrix.feature }}
+      - if: ${{ matrix.layout != 'none' && matrix.feature != 'none' }}
+        run: |
+          cargo build --no-default-features --features ${{ matrix.layout }},${{ matrix.feature }}
+          cargo test --no-default-features --features ${{ matrix.layout }},${{ matrix.feature }}
```

Comment:
I find all this branching a bit clunky but it was the only way I've found so far to be able to specify "nothing". Maybe we don't need to test things like `--no-default-features --features alloc`?

Reply:
I think the `serde` feature is possibly the one that ought to be pulled out into it's own job(s). I think `alloc` definitely needs to be tested. If anything it's the combinations that don't have either `std` or `alloc` that we could drop if want. I'm not sure how much sense it really makes to use Taffy in an environment without allocation.

Reply:
I've put the allocation features: `std` & `alloc` into its own axis in the matrix. It will now always test with either of them. 
I've also extracted `cargo test --no-default-features` into its own job.

Let me know what you think of the current setup